src/change_stream/mod.rs

```diff
@@ -0,0 +1,172 @@
+//! Contains the functionality for ChangeStreams.
+pub mod document;
+pub mod option;
+
+use std::{marker::PhantomData, pin::Pin, task::Poll};
+
+use bson::Document;
+use futures::{stream::Stream, task::Context};
+use serde::{de::DeserializeOwned, Deserialize};
+
+use crate::{
+    change_stream::{
+        document::{ChangeStreamEventDocument, ResumeToken},
+        option::ChangeStreamOptions,
+    },
+    error::Result,
+    options::AggregateOptions,
+    selection_criteria::ReadPreference,
+    Client,
+    Collection,
+    Cursor,
+    Database,
+};
+
+/// A `ChangeStream` streams the ongoing changes of its associated collection, database or
+/// deployment. `ChangeStream` instances should be created with method `watch` against the relevant
+/// target.
+/// `ChangeStream`'s are "resumable", meaning that they can be restarted at a certain place in the
+/// stream of events. This is done automatically when the ChangeStream encounters certain
+/// "resumable" errors, such as transient network failures. It can also be done manually by passing
+/// a ResumeToken retrieved from a past event into either the resume_after or start_after (4.2+)
```

Comment:
ditto backticks for `resume_after` and `start_after`, and can we make them links to the appropriate fields in the struct they live in?

Reply:
fixed